src/routes/assessmentRoutes.js

```diff
@@ -71,84 +106,139 @@ router.get("/assessments/:studentId", isLoggedIn, async (req, res) => {
       averageScore: (totalScores.averageScore / totalScores.count).toFixed(1),
     };
 
-    // Respond with the evaluations and their averages
+    // Render the assessments page
     res.render("assessment_student", {
       currentUser: req.user,
       student,
       evaluations,
       averages,
+      classId,
     });
   } catch (e) {
     console.error(e);
     res.status(500).json({ error: e.message });
   }
 });
 
+
 //Detailed Assessments
 router.get(
-  "/assessments-detailed",
+  "/assessments-detailed/:classId",
   isLoggedIn,
   isInstructor,
   async (req, res) => {
     try {
-      // Fetch all students
-      const students = await User.find({ user_type: "student" });
+      const { classId } = req.params;
+
+      // Validate the class exists
+      const currentClass = await Class.findById(classId);
+      if (!currentClass) {
+        return res.status(404).send("Class not found");
+      }
+
+      // Fetch unique evaluatees from evaluations for the class
+      const evaluations = await Evaluation.find({ class_id: classId });
+      const studentIds = [...new Set(evaluations.map((eval) => eval.evaluatee))];
 
-      // Fetch evaluations for each student without populating evaluator
+      // Fetch student details for those evaluatees
+      const students = await User.find({
+        _id: { $in: studentIds },
+        user_type: "student",
+      });
+
+      // Fetch evaluations for each student
```

Comment:
Add a check to ensure students is not empty before proceeding to fetch evaluations.